File: ios/MobilePassSDK/Model/PassFlowState.swift
//
//  PassFlowState.swift
//  MobilePassSDK
//
//  Created by Erinc Cakir on 1.06.2023.
//

import Foundation

public struct PassFlowState {
    public var state:       Int
    public var data:        String?
    public var datetime:    Date?
    
    init(state: PassFlowStateCode, data: String?, datetime: Date? = nil) {
        self.state      = state.rawValue
        self.data       = data
        self.datetime   = datetime
    }
}

File: android/mobilepasssdk/src/main/java/com/armongate/mobilepasssdk/model/PassFlowState.java
package com.armongate.mobilepasssdk.model;

import java.util.Date;

public class PassFlowState {

    public Integer  state;
    public String   data;
    public Date datetime;

    public PassFlowState(Integer state, String data) {
        this(state, data, new Date());
    }

    public PassFlowState(Integer state, String data, Date datetime) {
        this.state = state;
        this.data = data;
        this.datetime = datetime;
    }

    public Integer getState() {
        return state;
    }

    public String getData() {
        return data;
    }

    public Date getDatetime() {
        return datetime;
    }
}

